src/nvim/eval.c

```diff
@@ -24033,6 +24050,7 @@ static inline TerminalJobData *common_job_init(char **argv,
   }
   Process *proc = (Process *)&data->proc;
   proc->argv = argv;
+  proc->quote_argv = quote_argv;
```

Comment:
> How about making if (!proc->quote_argv && *p_sxq != NUL) { in libuv_process.c

Its kind of strange (from a logic perspective) to make a decision based on p_sxq so far from the reason behind it i.e. 6 months down the road if someone looks at the code in libuv_process.c and sees that what would they think?